public/push.php

```diff
@@ -3,11 +3,17 @@
 
 require_auth();
 
-if (empty($_FILES['package'])) {
+if (count($_FILES)==0)
+{
 	api_error('400', 'No package file');
 }
 
-$upload_filename = $_FILES['package']['tmp_name'];
```

Comment:
Does something like this work?

```
$file = head($_FILES);
```

Then you should be able to do `$file['tmp_name']` to get the filename.